homeassistant/config_entries.py

```diff
@@ -265,9 +266,35 @@ class ConfigFlowResult(FlowResult, total=False):
 
     minor_version: int
     options: Mapping[str, Any]
+    subentries: Mapping[str, ConfigSubentryData]
     version: int
 
 
+class ConfigSubentryData(TypedDict):
+    """Container for configuration subentry data."""
+
+    data: Mapping[str, Any]
+    title: str
+
+
+class SubentryFlowResult(FlowResult, total=False):
+    """Typed result dict for subentry flow."""
+
+    subentry_id: str
+
+
+@dataclass(frozen=True)
+class ConfigSubentry:
+    """Container for a configuration subentry."""
+
+    data: MappingProxyType[str, Any]
+    title: str
+
```

Comment:
Let's add an ID. Subentries will be used to set up entities, and this will allow using it for unique ID.